### ISSUE TITLE
Provide documentation for several versions + develop

### ISSUE THREAD

**Issue by BenjaminRodenberg** [precice/precice.github.io#14]
I think this would be a useful feature for developers as well as users of older preCICE versions. I know that this is not simple to implement and it will probably require to have a release cycle for `precice/precice.github.io` that is synchronized with `precice/precice`. We will also need a `master` and `develop` branch + tags corresponding to releases.

Here is an example: https://docs.bokeh.org/en/latest/ The dropdown menu at the top left allows to select the version of the documentation.

**Comment by chlorenz** [comment]
I did some research on versioning going into the development of the new documentation + website, and I'll share my thoughts behind our decision not to implement it at this time and when it might become appropriate to do so.

The TL;DR is versioning adds complexity, and as long as it is nice-to-have but not must-have, we should not implement it.

### Technical considerations
As you have hinted at, there is no easy way to do this [manually](https://xkcd.com/979/).
That's why versioning is billed as a ["killer feature"](https://docusaurus.io/docs/en/tutorial-version) of Docusaurus. Similarly Sphinx, the Python-based framework powering the Bokeh docs, supports versioning.
Using such a framework not only requires an upfront investment into an additional tech stack (JS/React or Python), but also leads to separation between docs and the rest of the website. In addition, there is recurring effort needed to manage every release.

### Do we need it?
There is no doubt that versioning is nice to have. So why does a unicorn, dev-oriented solution like Docker [not support it](https://docs.docker.com/docsarchive/)? In my opinion the crucial questions are:
1. Do we want to have users on older versions?
2. Do we have users on older versions?

Luckily the answer to both is no (and in my opinion it's a major boon, ask a native Android/iOS/Mac developer).

### When we might need it
A useful test is: If we had X hours of dev-time to advance preCICE, where would it be best spent? If we are at a point where implementing and maintaining versioning will contribute more to preCICE than improving the docs themselves, or implementing other useful features for preCICE, or fixing bugs, then the time has come to implement versioning.
Another factor are environmental changes: If, for whatever reason, we are required to maintain a [LTS](https://en.wikipedia.org/wiki/Long-term_support) version of preCICE, say, it also makes sense to have versioning.

In the meantime, older versions of the docs can always be built locally (the Docker approach) and we also provide pdfs for older versions (coming soon).

**Comment by MakisH** [comment]
tl;dr: very high effort, low benefit (or even accidental trouble).

As discussed now a few times, unless we make foundational changes (e.g., use another documentation system), this would be very complicated to implement, which we do not actively support old versions. We might also get users looking into older documentation by accident. I would prefer to know when new information does not work, rather than struggling to get the new information out.

Another consideration: our documentation concerns many components. We could refer to these using the distribution version, but I would argue this is now what we are doing anyway: In #653, I added more details and links to the web archive for various states in https://precice.org/fundamentals-previous-versions.html This solves the problem of users of older versions to a large extent, even if not completely.

Closing for now, feel free to reopen if there is a clear suggestion on how to implement something like this.
